src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -135,20 +135,41 @@ public IPAddress(byte[] address, long scopeid)
                 throw new ArgumentException(SR.dns_bad_ip_address, nameof(address));
             }
 
+            // Consider: Since scope is only valid for link-local and site-local
+            //           addresses we could implement some more robust checking here
+            if (scopeid < 0 || scopeid > 0x00000000FFFFFFFF)
+            {
+                throw new ArgumentOutOfRangeException(nameof(scopeid));
+            }
```

Comment:
FYI: the diff here is a bit hard to read... all I did was move the arg validation up, as there's no point in allocating the ushort[] if the argument validation fails